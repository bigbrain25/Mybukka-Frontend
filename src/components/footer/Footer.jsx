import React from 'react';
import UrlLists from '../list/UrlLists';

import './footer.scss';

const customUrlListsProps = {
  links: [
    { text: 'link 1', href: '/', key: '1' },
    { text: 'link 2', href: '/', key: '2' },
    { text: 'link 3', href: '/', key: '3' },
    { text: 'link 4', href: '/', key: '4' },
    { text: 'link 5', href: '/', key: '5' },
    { text: 'link 6', href: '/', key: '6' }
  ]
};

const Footer = () => (
  <div className="footer">
    <div className="container">
      <div className="row navs">
        <div className="col col-12 col-sm-12 col-md-6 col-lg-3 col-xl-2 footer-brand-section">
          <h4 className="footer-brand-name">Mybukka</h4>
        </div>
        <div className="col col-6 col-sm-6 col-md- 6 col-lg-3 col-xl-2 nav-section">
          <h4 className="nav-title">MYBUKKA</h4>
          <UrlLists
            links={customUrlListsProps.links}
            classNames="list-item-white"
          />
        </div>
        <div className="col col-6 col-sm-6 col-md- 6 col-lg-3 col-xl-2 nav-section">
          <h4 className="nav-title">CONTACT US</h4>
          <UrlLists
            links={customUrlListsProps.links}
            classNames="list-item-white"
          />
        </div>
        <div className="col col-6 col-sm-6 col-md- 6 col-lg-3 col-xl-2 nav-section">
          <h4 className="nav-title">SOCIAL MEDIA</h4>
          <UrlLists
            links={customUrlListsProps.links}
            classNames="list-item-white"
          />
        </div>
        <div className="col col-6 col-sm-6 col-md- 6 col-lg-3 col-xl-2 nav-section">
          <h4 className="nav-title">SOCIAL MEDIA</h4>
          <UrlLists
            links={customUrlListsProps.links}
            classNames="list-item-white"
          />
        </div>
        <div className="col col-6 col-sm-6 col-md- 6 col-lg-3 col-xl-2 nav-section">
          <h4 className="nav-title">SOCIAL MEDIA</h4>
          <UrlLists
            links={customUrlListsProps.links}
            classNames="list-item-white"
          />
        </div>
      </div>
      <div className="copyright-text text-center nav-section">
        <p>Â©2019 MYBUKKA INC TERMS PRIVACY</p>
      </div>
    </div>
  </div>
);

export default Footer;
