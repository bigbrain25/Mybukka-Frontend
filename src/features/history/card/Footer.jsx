import React from 'react';
import Row from 'Components/grid/Row';
import PropTypes from 'prop-types';

const Footer = ({ status, price }) => {
  const colorObj = {
    pending: 'warning',
    delivered: 'success',
    cancelled: 'danger'
  };

  return (
    <Row classNames="justify-content-between text-capitalize">
      <h5 className="font-size-14">Status: <span className={`text-${colorObj[status]}`}> {status}</span></h5>
      <h5 className="text-color font-size-14"> â‚¦{price}.00</h5>
    </Row>
  );
};

export default Footer;

Footer.propTypes = {
  status: PropTypes.string.isRequired,
  price: PropTypes.string.isRequired
};
