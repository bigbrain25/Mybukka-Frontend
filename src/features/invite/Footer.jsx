import React, { useContext } from 'react';
import PropTypes from 'prop-types';
import Twitter, { Facebook } from 'Components/button/SocialSvg';
import { CopyToClipboard } from 'react-copy-to-clipboard';
import MyContextPush from './context-api/MyContextPush';
import Button from '../../components/button/Button';
import './InviteFriends.css';

const Footer = ({ inputData, handleCopy }) => {
  const props = useContext(MyContextPush);
  return (
    <section>
      <div className="d-flex flex-column">
        <div className="mx-4">
          <CopyToClipboard text={inputData.link} onCopy={() => handleCopy()}>
            <div className="cursor-pointer invite-link invite-padding
            d-flex justify-content-around align-items-center"
            >
              {inputData.link}
              <span
                className={inputData.copied ?
                  'text-success' : 'text-warning'}
              >{inputData.copied ? 'COPIED' : 'COPY' }</span>
            </div>
          </CopyToClipboard>
        </div>
        <div className="social invite-padding
        d-flex flex-row justify-content-end mx-4"
        >
          <button
            className="facebook-btn mb-4 w-50
            mr-4 d-flex justify-content-around"
            type="button"
            onClick={() => props.push('/social/facebook')}
          >
            <Facebook />
            <span className="text-white">SHARE</span>
          </button>
          <Button
            classNames="twitter-btn mb-4 w-50 mr-4 d-flex justify-content-around"
            type="button"
            handleClick={() => props.push('/social/twitter')}
          >
            <Twitter />
            <span className="text-white">SHARE</span>
          </Button>
        </div>
      </div>
    </section>
  );
};

export default Footer;

Footer.propTypes = {
  inputData: PropTypes.objectOf(
    PropTypes.oneOfType([
      PropTypes.string,
      PropTypes.number,
      PropTypes.bool,
    ])).isRequired,
  handleCopy: PropTypes.func.isRequired,
  push: PropTypes.func.isRequired,
};
