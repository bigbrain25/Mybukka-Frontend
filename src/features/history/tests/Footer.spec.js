import React from 'react';
import Footer from '../card/Footer';

describe('Card Footer component', () => {
  const props = {
    price: '2000.98',
    status: 'delivered'
  };

  const wrapper = mount(<Footer {...props} />);

  it('renders properly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
