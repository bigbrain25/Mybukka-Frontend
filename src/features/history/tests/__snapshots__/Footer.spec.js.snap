// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Footer component renders properly 1`] = `ReactWrapper {}`;
